services/frontend-v3/src/components/diplomaTable/DiplomaTable.jsx

```diff
@@ -0,0 +1,172 @@
+  const [reset, setReset] = useState(false);
+  const [searchText, setSearchText] = useState("");
+  const [searchedColumn, setSearchedColumn] = useState("");
+  const [selectedRowKeys, setSelectedRowKeys] = useState([]);
+  const [size] = useState("large");
+
+  const { type } = props;
+
+  useEffect(() => {
+    let diplomas = [];
+    if (loading) {
+      const setState = async () => {
+        diplomas = await getDiplomas();
+        setData(diplomas);
+        setLoading(false);
+      };
+      setState();
+    } else {
+      const updateState = async () => {
+        diplomas = await getDiplomas();
+        setData(diplomas);
+        setReset(false);
+      };
+      updateState();
+    }
+  }, [loading, reset]);
+
+  const getDiplomas = async () => {
+    try {
+      let results = await axios.get(
+        backendAddress + "api/admin/options/" + type
+      );
+      return results.data;
+    } catch (error) {
+      console.log(error);
+      return 0;
+    }
+  };
+
+  const getDisplayType = (plural) => {
+    if (plural)
+      return props.intl.formatMessage({
+        id: "admin." + type + ".plural",
+        defaultMessage: type,
+      });
+
+    return props.intl.formatMessage({
+      id: "admin." + type + ".singular",
+      defaultMessage: type,
+    });
+  };
+
+  const handleSearch = (selectedKeys, confirm, dataIndex) => {
+    confirm();
+    setSearchText(selectedKeys[0]);
+    setSearchedColumn(dataIndex);
+  };
+
+  const handleReset = (clearFilters) => {
+    clearFilters();
+    setSearchText("");
+  };
+
+  const handleSubmitAdd = async (values) => {
+    try {
+      const url = backendAddress + "api/admin/options/" + type;
+
+      await axios.post(url, {
+        descriptionEn: values.addDiplomaEn,
+        descriptionFr: values.addDiplomaFr,
+      });
+
+      setReset(true);
+    } catch (error) {
+      console.log(error);
+      return 0;
+    }
+  };
+
+  const handleSubmitEdit = async (values, id) => {
+    try {
+      const url = backendAddress + "api/admin/options/" + type + "/" + id;
+
+      await axios.put(url, {
+        descriptionEn: values.editDiplomaEn,
+        descriptionFr: values.editDiplomaFr,
+      });
+
+      setReset(true);
+    } catch (error) {
+      console.log(error);
+      return 0;
+    }
+  };
+
+  const handleSubmitDelete = async () => {
+    try {
+      const url = backendAddress + "api/admin/delete/" + type;
+
+      await axios.post(url, { ids: selectedRowKeys });
+
+      setSelectedRowKeys([]);
+      setReset(true);
+    } catch (error) {
+      console.log(error);
+      return 0;
+    }
+  };
+
+  const rowSelection = {
+    onChange: (selectedRowKeys) => {
+      onSelectChange(selectedRowKeys);
+    },
+  };
+
+  const onSelectChange = (selectedRowKeys) => {
+    // console.log("selectedRowKeys changed: ", selectedRowKeys);
+    setSelectedRowKeys(selectedRowKeys);
+  };
+
+  const convertToViewableInformation = () => {
```

Comment:
please comment the purpose of each function above it

Reply:
Yeah, need to comment on some other things from Admin side too.

Reply:
I am thinking to do all the commenting on one branch and then PR it?

Reply:
i would do as much commenting as you can related to this PR. in case your next PR gets reverted for some reason. Its just best practice.